src/main/java/com/uid2/client/IdentityMapInput.java

```diff
@@ -0,0 +1,88 @@
+package com.uid2.client;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class IdentityMapInput {
+    /**
+     * @param emails a list of normalized or unnormalized email addresses
+     * @return a IdentityMapInput instance, to be used in {@link IdentityMapHelper#createEnvelopeForIdentityMapRequest}
+     */
+    public static IdentityMapInput fromEmails(Iterable<String> emails) {
+        return new IdentityMapInput(IdentityType.Email, emails, false);
+    }
+
+    /**
+     * @param phones a <a href="https://unifiedid.com/docs/getting-started/gs-normalization-encoding#phone-number-normalization">normalized</a> phone number
+     * @return an IdentityMapInput instance
+     */
+    public static IdentityMapInput fromPhones(Iterable<String> phones) {
+        return new IdentityMapInput(IdentityType.Phone, phones, false);
+    }
+
+    /**
+     * @param hashedEmails a <a href="https://unifiedid.com/docs/getting-started/gs-normalization-encoding#email-address-normalization">normalized</a> and <a href="https://unifiedid.com/docs/getting-started/gs-normalization-encoding#email-address-hash-encoding">hashed</a> email address
+     * @return an IdentityMapInput instance
+     */
+    public static IdentityMapInput fromHashedEmails(Iterable<String> hashedEmails) {
+        return new IdentityMapInput(IdentityType.Email, hashedEmails, true);
+    }
+
+    /**
+     * @param hashedPhones a <a href="https://unifiedid.com/docs/getting-started/gs-normalization-encoding#phone-number-normalization">normalized</a> and <a href="https://unifiedid.com/docs/getting-started/gs-normalization-encoding#phone-number-hash-encoding">hashed</a> phone number
+     * @return an IdentityMapInput instance
+     */
+    public static IdentityMapInput fromHashedPhones(Iterable<String> hashedPhones) {
+        return new IdentityMapInput(IdentityType.Phone, hashedPhones, true);
+    }
+
+    private IdentityMapInput(IdentityType identityType, Iterable<String> emailsOrPhones, boolean alreadyHashed) {
```

Comment:
What happens if `emailsOrPhones` is empty?

Reply:
The request still works but getMappedIdentities() and getUnmappedIdentities return empty maps. Added test `identityMapEmptyInput`